### ISSUE TITLE
Feature request: Better vertices and tangents handling

### ISSUE THREAD

**Issue by yoSachinkr** [synfig/synfig#2644]
As of now, working with vertices and tangents is a bit of a chore in synfig. It drastically slows down the process of creating artwork within the application. Function like splitting the tangents radius/angle on the fly while creating paths, which are present in other vector based apps e.g. inkscape, is non existent in synfig. Rather it requires an extra step of opening the context menu and selecting the split tangents option. ( imagine doing this over and over again with hundreds of vertices in a complex artwork and i'm sure you can feel the pain too)
Another issue that must be sorted is when you create a point (vertex) with zero tangent radius, and then later if you want to bring out those tangent handles, there's no easy way to do so. Either you'll have to fiddle with the path adjacent to the vertex to get those handle pop out or go to parameters panel to manually put values to the tangent radius. (hovering the mouse over the vertex does nothing to indicate the tangent handles.
So I'd request devs to make vertices and tangents handling a bit more in line with other vector based apps as it would help everyone tremendously.

**Comment by Gilraiser** [comment]
Hello, cordial greeting. Looking in other programs I find these options:

![node-options-3](https://user-images.githubusercontent.com/45057390/169926166-9114edc8-fba9-4436-9d6f-91e1a7f233dd.png)
![node-options-2](https://user-images.githubusercontent.com/45057390/169926168-fdb9ec44-27dc-4179-aef8-c6ebd1e52409.png)
![node-options](https://user-images.githubusercontent.com/45057390/169926170-dd7d49ad-d301-436e-b9eb-b92589c9c791.png)

The truth is that the editing of vector graphics in Synfig seems to me to be one of the best I've seen. However it would be a great improvement if the size of the handles could be increased and if the options of the images that I attached above were in the transformation tool options.

![node-options-final](https://user-images.githubusercontent.com/45057390/169926816-cac5f69f-11d8-4213-ba93-34ac3370e216.png)

**Comment by yoSachinkr** [comment]
In my opinion, GIMP's way of handling vertices and tangents seems quite intuitive to me.

https://user-images.githubusercontent.com/50204208/169980567-3778b3e1-916f-49dc-888b-a47a01b29119.mp4

If you observe this video carefully, you'll notice that when i create a vertex and drag the mouse, the tangent handles drag equally in opposite directions, then when i lift the mouse and grab the tangent handle to move, i can do it easily in any direction as the tangent is split by default. But when I shift+click on the tangent, the tangent angle is joined as long as I hold shift key and manipulate the tangent handle. To complete the loop, I have to ctrl + click the first vertex.

In the whole process, I had never have to open the context menu even for once.

A behaviour similar to this would definitely ease out the process of art creation in synfig.

**Comment by rodolforg** [comment]
> Another issue that must be sorted is when you create a point (vertex) with zero tangent radius, and then later if you want to bring out those tangent handles, there's no easy way to do so.

Simpler than your options is this way:
![Peek 29-05-2022 11-04](https://user-images.githubusercontent.com/1323987/170873144-555a6411-ee16-45c5-a424-f5b5a4941bc9.gif)

:)

A while ago I was thinking about implement something like  https://bezier.method.ac/ and https://helpx.adobe.com/photoshop/using/editing-paths.html

**Comment by yoSachinkr** [comment]
@rodolforg I've used the adobe's pen tool, but to be honest, I think GIMP has nailed the pen tool implementation. I find it very intuitive and handy.

**Comment by rodolforg** [comment]
Did you see bezier game I linked? :)

Gimp vs Synfig is all about tangents?

**Comment by yoSachinkr** [comment]
> Did you see bezier game I linked? :)

Yes, I did. 😀 It's just like using pen tool in photoshop/inkscape.


> Gimp vs Synfig is all about tangents?

Obviously not. Both have different use cases. But when it comes to path handling, both the apps have different approaches. And these differences can be felt while using pen tool in each of them.
Since, in synfig, path creation is one of the most import part of creating a composition, so I felt the need to put this issue forth. I did some preliminary research and found GIMP's handling of paths much more intuitive as you can see in the video above. In fact, I'd request you to try it yourself and I'm sure you'll appreciate it too.

**Comment by rodolforg** [comment]
> Obviously not.

Sorry, I meant when we are drawing path. :)

The difference is that you can split tangents after drawing a path segment and adjust bézier while drawing.
- Synfig doesn't allow bézier handling with Bline Tool, but Transform tool do
- Synfig doesn't allow to split tangents with Bline Tool "directly" (only with context menu it does), but Transform Tool do (by pressing `shift`)

Synfig also misses how to continue to draw from the selected vertex by using the Bline Tool.

Anything else?

**Comment by Gilraiser** [comment]
Hello, I think @yoSachinkr idea is good and it would be a good idea if there were also keyboard shortcuts at the moment the curve is being drawn to divide tangents. There are already keyboard shortcuts to split tangents (shift + click on tangent handle), but when the path is already finished with the transform tool.

**Comment by yoSachinkr** [comment]
I think that's it. 🤔
